### PR TITLE
Error handling

### PR DESCRIPTION
**Pull request by shaharariel95**

### CODE REVIEW

File: client/homePage.html [hunk 1]
```diff
@@ -63,7 +63,7 @@ <h3 class="welcome-sub-text">Welcome to the IAM project </h3>
             <a class="dropdown-item"><i
                     class="bi bi-person-circle dropdown-item-icon mdi mdi-account-outline text-primary me-2"></i>
               My Profile</a>
-            <a class="dropdown-item"><i
+            <a class="dropdown-item" id='logout'><i
```
Comment (Ronina24) [inline client/homePage.html:66]: :smiley:

File: client/script/script.js [hunk 2]
```diff
@@ -332,6 +333,23 @@ function MenuPermission() {
...
+  };
+  console.log(data);
+  const response = await fetch(`${runningPath}/auth/logout`, {
+    method: 'POST',
+    headers: {
+      'Content-Type': 'application/json',
+    },
+    body: JSON.stringify(data),
+  });
+  // const body = await response.json();
```
Comment (Ronina24) [inline client/script/script.js:348]: no comments please :(

File: middleware/authenticate.js [hunk 3]
```diff
@@ -49,15 +48,14 @@ const refreshTokenVerify = async (req, res) => {
 };
 
 const authenticateToken = async (req, res, next) => {
-  console.log(req.cookies.jwt);
+  console.log('checked token');
 
   const authHeader = req.headers.authorization;
   const token = authHeader && authHeader.split(' ')[1];
-  // if (token == null) return res.sendStatus(401);
+  // if (token == null) return res.redirect('/');
```
Comment (Ronina24) [inline middleware/authenticate.js:55]: comments :imp:

File: middleware/validatorService.js [hunk 4]
```diff
@@ -54,14 +53,14 @@ const checkPermission = async (req, res, next) => {
...
-    const user = await User.retrieve(req.body.mail);
     if (user.type === 'user' || user.type === 'manager') throw new Error('Not authorized');
-    // check sit
-  }
-  catch(err){
-    res.status(403);
+  } catch (e) {
+    console.log(e);
   }
+  // check sit
```
Comment (Ronina24) [inline middleware/validatorService.js:62]: COMMENTS!!!!!!! :imp: :imp:

File: client/script/script-auth.js [hunk 5]
```diff
@@ -55,11 +55,11 @@ const login = async () => {
     },
     body: JSON.stringify(data),
   });
-  const body = await response.json();
-  if (response.status !== 200 && body.message) {
-    alert((body.message));
+  // const body = await response.json();
```
Comment (dkracheli) [inline client/script/script-auth.js:58]: Delete comments 
:innocent:

File: models/Account.model.js [hunk 6]
```diff
@@ -8,7 +8,7 @@ const accountSchema = new Schema({
   assets: {
     credits: { type: Number, default: 0 },
     seats: { type: Number, default: 0 },
-    usedSeats: { type: String, default: 0 },
+    usedSeats: { type: Number, default: 0 },
```
Comment (Ronina24) [inline models/Account.model.js:11]: great! :innocent:

File: client/script/script.js [hunk 7]
```diff
@@ -332,6 +333,23 @@ function MenuPermission() {
 
 window.addEventListener('load', sideMenu);
 
+Logout.addEventListener('click', async () => {
```
Comment (dkracheli) [inline client/script/script.js:336]: Nice job :stuck_out_tongue_winking_eye:

File: controllers/accountsController.js [hunk 8]
```diff
@@ -1,14 +1,13 @@
...
 const inviteUser = async (req, res) => {
   try {
-    // checkPermission(req.user);
     const manager = await User.retrieve({ email: req.user });
     const account = await Account.retrieve({ _id: manager.accountId });
     const invitedUser = await User.retrieve({ email: req.params.email });
     if (invitedUser) {
-      if (account._id.toString() === invitedUser.accountId) throw new Error('user alredy......');
+      if (account._id.toString() === invitedUser.accountId) throw new Error('user already......');
       await sendInvitation(req.params.email, invitedUser);
```
Comment (dkracheli) [inline controllers/accountsController.js:11]: r('user already......') ??????

:confused::confused::confused::confused::confused:

**Review by Ronina24 — COMMENTED** [review]
great work! only bud thing is comments comments and more comments

File: controllers/accountsController.js [hunk 9]
```diff
@@ -92,8 +89,8 @@ const editAccount = async (req, res) => {
 const disableAccount = async (req, res) => {
   if (!req.body.name) { res.status(401); }
   const acc = Account.find({ name: req.body.name });
-  const users = User.find({ accountId: acc._id });
-  // users.i;
+  User.deleteMany({ accountId: acc._id });
+  Account.update({ _id: acc._id }, { status: 'disabled' });
 };
```
Comment (dkracheli) [inline controllers/accountsController.js:94]: :muscle::muscle::muscle:

File: controllers/registerController.js [hunk 10]
```diff
@@ -37,9 +37,10 @@ const handleConfirmCode = async (req, res) => {
     if (user) throw new Error('user already exist');
     const oneTimePassRecord = await existCode(req.body.email);
     await register.otpCompare(req.body.code, oneTimePassRecord.code);
-    const accountID = await createAccount(req.body.email);
-    await register.createUser(req.body, accountID);
-    return res.status(200)
+    const account = await createAccount(req.body.email);
```
Comment (dkracheli) [inline controllers/registerController.js:40]: It works?

**Review by dkracheli — COMMENTED** [review]
Good job

**Pull request merged**